src/bin/ambit/directories.rs

```diff
@@ -44,7 +44,17 @@ impl AmbitPath {
     pub fn as_string(&self) -> AmbitResult<String> {
         match self.kind {
             AmbitPathKind::File => {
-                let mut file = File::open(&self.path)?;
+                let mut file = match File::open(&self.path) {
+                    Ok(file) => file,
+                    Err(e) => {
+                        // Format error nicely to notify user of error from file open attempt
+                        return Err(AmbitError::Other(format!(
+                            "Failed to read `{}`\n\nCaused by:\n  {}",
```

Comment:
It may be better to create a general formatting for causes in `impl Display for AmbitError`. See [the Error trait, more specifically the `source()` method.](https://doc.rust-lang.org/std/error/trait.Error.html#method.source)

Reply:
Should I implement an error struct like this (with `Error` and `Display` traits implemented)?

```rust
#[derive(PartialEq, Eq, Debug)]
pub struct AmbitPathError {
    reason: io::Error;
}
```
If so, should I define it in `src/error.rs`?

Reply:
Note that `Error::source()` yields an `Option<_>`. It is therefore possible to implement the `source()` function for `AmbitError` (and have it return `None` if there is no source), and in `impl Display for AmbitError` format the cause correctly. This would also help with `AmbitError::Io`.